### DISCUSSION TITLE
Custom Nginx Configuration??

### DISCUSSION THREAD

**Discussion by asad-awadia** [nginx-proxy/nginx-proxy#1671] [Q&A]
```
If you need to configure Nginx beyond what is possible using environment variables, you can provide custom configuration files on either a proxy-wide or per-VIRTUAL_HOST basis.
```

What does this mean exactly? I get conf for global stuff i.e. things that apply to all servers but how do I add specific things to specific servers like rate limiting [or gzip] using limit_req etc

Does the conf file need to be in a specific format for it to be included? I read through the nginx tmpl file and can't find how it's being included

Tried going to the google group to ask the question but getting

```
This group either doesn't exist, or you don't have permission to access it. If you're sure this group exists, contact the Owner of the group and ask them to give you access.
```

**Comment by DatAres37** [comment]
You can add configurations per Virtual Host: https://github.com/jwilder/nginx-proxy#per-virtual_host

**Comment by sgabe** [comment]
I have summarized the various options in https://github.com/nginx-proxy/nginx-proxy/issues/1398#issuecomment-587717134, it may help to figure it out.
